File: leetcode/algo2/day4/844_backspace_string_compare.js
var backspaceCompare = function (s, t) {
  let ss = "",
    tt = "";
  for (let i = 0; i < s.length; i++) {
    if (s[i] !== "#") {
      ss = ss.concat(s[i]);
    } else if (ss.length) {
      ss = ss.slice(0, ss.length - 1);
    }
  }
  for (let i = 0; i < t.length; i++) {
    if (t[i] !== "#") {
      tt = tt.concat(t[i]);
    } else if (tt.length) {
      tt = tt.slice(0, tt.length - 1);
    }
  }
  console.log(ss, tt);
  return ss === tt;
};

var s = "ab###";
var t = "c#d#";
console.log(backspaceCompare(s, t));

File: leetcode/algo2/day4/844_backspace_string_compare.cpp
#include <string>
#include <iostream>
bool backspaceCompare(std::string s, std::string t)
{
    std::string ss = "", tt = "";
    for (int i = 0; i < s.size(); i++)
    {
        if (s[i] != '#')
            ss.push_back(s[i]);
        else if (ss.size())
            ss.pop_back();
    }

    for (int i = 0; i < t.size(); i++)
    {
        if (t[i] != '#')
            tt.push_back(t[i]);
        else if (tt.size())
            tt.pop_back();
    }

    return ss == tt;
}
int main()
{
    std::string s = "ab###", t = "c#d#";
    std::cout << backspaceCompare(s, t) << std::endl;
}
